fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraVersioningIT.java

```diff
@@ -989,52 +926,35 @@ public void testCreateVersionOfBinary() throws Exception {
         }
     }
 
-    @Ignore("get memento not implemented")
     @Test
-    public void testCreateVersionOfBinaryWithDatetimeAndContentType() throws Exception {
+    public void testCreateVersionOfBinaryWithDatetimeAndBody() throws Exception {
         createVersionedBinary(id);
 
         final String mementoUri = createMemento(subjectUri);
+        final String v1Time = now();
         assertMementoUri(mementoUri, subjectUri);
 
-        // Verify that the memento has the updated binary
-        try (final CloseableHttpResponse response = execute(new HttpGet(mementoUri))) {
-            assertMementoDatetimeHeaderMatches(response, MEMENTO_DATETIME);
+        TimeUnit.SECONDS.sleep(1);
 
-            assertEquals("Binary content of memento must be empty",
-                    "", EntityUtils.toString(response.getEntity()));
-            assertEquals(OCTET_STREAM_TYPE, response.getFirstHeader(CONTENT_TYPE).getValue());
-        }
-    }
+        putVersionedBinary(id, OCTET_STREAM_TYPE, BINARY_UPDATED, true);
 
-    @Ignore("get memento not implemented")
-    @Test
-    public void testCreateVersionOfBinaryWithBody() throws Exception {
-        createVersionedBinary(id);
+        final String mementoUri2 = createMemento(subjectUri);
+        final String v2Time = now();
+        assertMementoUri(mementoUri2, subjectUri);
 
-        final String mementoUri = createMemento(subjectUri);
-        assertMementoUri(mementoUri, subjectUri);
+        // Verify that the memento has the updated binary
+        try (final CloseableHttpResponse response = execute(new HttpGet(mementoUri))) {
+            assertMementoDatetimeHeaderMatches(response, v1Time);
 
-        final HttpGet httpGet = new HttpGet(mementoUri);
-        try (final CloseableHttpResponse response = execute(httpGet)) {
-            assertMementoDatetimeHeaderPresent(response);
+            // Content-type is not retained for a binary memento created without description
```

Comment:
I'm not sure what that means... It only looks like I added it because of other editing I did in the area.

https://github.com/fcrepo4/fcrepo4/blob/f02d2e637e6364db133bddbe759470e7a7c7d4d1/fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraVersioningIT.java#L1039

??

Reply:
I suppose we should ensure that the content-type for binary mementos is retained... then the comment can go away :)